agent-control/src/sub_agent/remote_config_parser.rs

```diff
@@ -73,22 +73,27 @@ where
 }
 
 /// Extracts the opamp remote configuration values and parses them to [YAMLConfig], if the values are empty it returns None.
+/// Check [crate::opamp::remote_config::AGENT_CONFIG_PREFIX] for multi configuration process behavior.
 fn extract_remote_config_values(
     opamp_remote_config: &OpampRemoteConfig,
 ) -> Result<Option<RemoteConfig>, RemoteConfigParserError> {
-    let remote_config_value = opamp_remote_config.get_default().map_err(|err| {
-        RemoteConfigParserError::InvalidValues(format!(
-            "could not load remote configuration values: {err}"
-        ))
-    })?;
-
-    if remote_config_value.is_empty() {
+    let config = opamp_remote_config.agent_configs_iter().try_fold(
```

Comment:
Considering that `agent_configs_iter` filter the configurations not starting with a prefix we could end up with the same configuration values for different hashes values, right?

Ex:

ConfigA --> `{"agentConfig": "{\"some-key\": \"some-value\"}", "ignored": "{}"}`
ConfigB --> `{"agentConfig": "{\"some-key\": \"some-value\"}", "also-ignored": "{}"}`

I don't think it is a problem because, considering the server's behavior, it would be a corner-case and the only consequence could be restarting some sub-agent with no actual configuration change. I'm not sure if it would be worth documenting 🤔